### PR TITLE
Refresh single volume when volume event emitted

### LINKED ISSUE #2245: support for volume labels
Volume labels were recently added to docker. See https://github.com/docker/docker/pull/21270

Creating  a label via docker to swarm.

```
docker volume create -d flocker --label owner=ryan --name test4 -o size=75G
```

Output 
```
$ docker volume inspect test4
[
    {
        "Name": "test4",
        "Driver": "flocker",
        "Mountpoint": "",
        "Labels": null
    }
]
```

Expected response 

```
$ docker volume inspect test4
[
    {
        "Name": "test4",
        "Driver": "flocker",
        "Mountpoint": "",
        "Labels": {
            "owner": "ryan"
        }
    }
]
```

The only way to see the label is to SSH to the node and run `inspect` on the local docker daemon.

```
docker-machine ssh mha-demo1
root@mha-demo1:~# docker volume inspect test4
[
    {
        "Name": "test4",
        "Driver": "flocker",
        "Mountpoint": "/flocker/8c22e40c-2519-4db6-9e9b-904bbbb6ea98",
        "Labels": {
            "owner": "ryan"
        }
    }
]
```

`docker version` to swarm from client
```
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:20 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      swarm/1.2.2
 API version:  1.22
 Go version:   go1.5.4
 Git commit:   34e3da3
 Built:        Mon May  9 17:03:22 UTC 2016
 OS/Arch:      linux/amd64
```

`docker version` from a docker host in the swarm cluster.
```
docker version
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:30:23 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Tue Apr 26 23:30:23 2016
 OS/Arch:      linux/amd64
```

### PR DESCRIPTION
**Pull request by allencloud**
This PR fixed part of issue #2245, not totally. 
Here is the reason:
1. When Swarm starts at the first time or restarts, Swarm will fetch all volumes via function `RefreshVolumes`, and it uses `/volumes` API against docker engine's which api misses volume's label vesus `/volumes/{ID}`.
2. When volumes updated and created during Swarm's running, Swarm will get infos about them via  `/volumes/{ID}`, which will not miss volume's labels.

This PR did:
1.Refresh single volume when volume event emitted. It will save network package size when dealing with volume.
2.Refresh **ONLY** the created volume when a volume is created.

This PR is similar to #2255 

Some related details:
https://github.com/docker/docker/issues/22838

Signed-off-by: allencloud <allen.sun@daocloud.io>

### CODE REVIEW

**Comment by doronp** [comment]
@allencloud I was looking at it as well. There is the exact same problem for networks by the way. I was not sure that full refresh every time is the way to go. In containers refresh for example we try to reduce that. 
PS I will open the issue about networks as well.

**Comment by allencloud** [comment]
@doronp Yes, same case.
API `/networks` against docker engine will miss field `Options` of each network, while `/networks/{*name}` has it.

**Comment by allencloud** [comment]
PTAL, similar to network
@dongluochen @abronan @nishanttotla

**Comment by abronan** [comment]
LGTM

ping @nishanttotla @dongluochen

**Comment by allencloud** [comment]
ping @nishanttotla @dongluochen

File: cluster/engine.go [hunk 1]
```diff
@@ -628,6 +628,28 @@ func (e *Engine) RefreshVolumes() error {
...
+			delete(e.volumes, IDOrName)
+			e.Unlock()
+			return nil
+		} else {
+			return err
+		}
+	}
+
+	e.Lock()
+	e.volumes[IDOrName] = &Volume{Volume: volume, Engine: e}
```
Comment (dongluochen) [inline cluster/engine.go]: I think we use volume name to index. It should be 
```
e.volumes[volume.Name] = &Volume{Volume: volume, Engine: e}
```

Comment (allencloud) [inline cluster/engine.go]: Thanks for your review @dongluochen 
Updated.

**Comment by dongluochen** [comment]
LGTM

**Pull request merged**